ESP32-P4 unable to build firmware with wifi enabled

Configuration should tell you you can't use wifi unless esp32_hosted is configured.

### The problem

```
In file included from src/esphome/components/wifi/wifi_component_esp_idf.cpp:9:
src/esphome/components/wifi/wifi_component_esp_idf.cpp: In member function 'void esphome::wifi::WiFiComponent::wifi_pre_setup_()':
/home/zarya/.platformio/packages/framework-espidf/components/esp_wifi/include/esp_wifi.h:303:26: error: 'CONFIG_ESP_WIFI_STATIC_RX_BUFFER_NUM' was not declared in this scope; did you mean 'CONFIG_ETH_DMA_RX_BUFFER_NUM'?
  303 |     .static_rx_buf_num = CONFIG_ESP_WIFI_STATIC_RX_BUFFER_NUM,\
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:177:28: note: in expansion of macro 'WIFI_INIT_CONFIG_DEFAULT'
  177 |   wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~
/home/zarya/.platformio/packages/framework-espidf/components/esp_wifi/include/esp_wifi.h:304:27: error: 'CONFIG_ESP_WIFI_DYNAMIC_RX_BUFFER_NUM' was not declared in this scope; did you mean 'WIFI_DYNAMIC_TX_BUFFER_NUM'?
  304 |     .dynamic_rx_buf_num = CONFIG_ESP_WIFI_DYNAMIC_RX_BUFFER_NUM,\
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:177:28: note: in expansion of macro 'WIFI_INIT_CONFIG_DEFAULT'
  177 |   wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~
/home/zarya/.platformio/packages/framework-espidf/components/esp_wifi/include/esp_wifi.h:305:20: error: 'CONFIG_ESP_WIFI_TX_BUFFER_TYPE' was not declared in this scope; did you mean 'CONFIG_ETH_DMA_TX_BUFFER_NUM'?
  305 |     .tx_buf_type = CONFIG_ESP_WIFI_TX_BUFFER_TYPE,\
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:177:28: note: in expansion of macro 'WIFI_INIT_CONFIG_DEFAULT'
  177 |   wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~
/home/zarya/.platformio/packages/framework-espidf/components/esp_wifi/include/esp_wifi.h:308:25: error: 'CONFIG_ESP_WIFI_DYNAMIC_RX_MGMT_BUF' was not declared in this scope
  308 |     .rx_mgmt_buf_type = CONFIG_ESP_WIFI_DYNAMIC_RX_MGMT_BUF,\
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:177:28: note: in expansion of macro 'WIFI_INIT_CONFIG_DEFAULT'
  177 |   wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~
/home/zarya/.platformio/packages/framework-espidf/components/esp_wifi/include/esp_wifi.h:323:31: error: 'CONFIG_ESP_WIFI_ESPNOW_MAX_ENCRYPT_NUM' was not declared in this scope
  323 |     .espnow_max_encrypt_num = CONFIG_ESP_WIFI_ESPNOW_MAX_ENCRYPT_NUM, \
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/wifi/wifi_component_esp_idf.cpp:177:28: note: in expansion of macro 'WIFI_INIT_CONFIG_DEFAULT'
  177 |   wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~
*** [.pioenvs/home-display/src/esphome/components/wifi/wifi_component_esp_idf.cpp.o] Error 1
=============================================================================== [FAILED] Took 21.61 seconds ===============================================================================
```

### Which version of ESPHome has the issue?

2025.8.2

### What type of installation are you using?

pip

### What platform are you using?

ESP32

### Component causing the issue

wifi

### YAML Config

```yaml
esphome:
  name: wifi-only-test
  friendly_name: wifi-only-test

esp32:
  board: esp32-p4
  framework:
    type: esp-idf


# Enable logging
logger:
  hardware_uart: UART0

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
```

### Anything in the logs that might be useful for us?

```txt

```

### Additional information

_No response_

--- a/esphome/components/wifi/__init__.py
+++ b/esphome/components/wifi/__init__.py
@@ -51,7 +51,7 @@ from . import wpa2_eap
 
 AUTO_LOAD = ["network"]
 
-NO_WIFI_VARIANTS = [const.VARIANT_ESP32H2]
+NO_WIFI_VARIANTS = [const.VARIANT_ESP32H2, const.VARIANT_ESP32P4]
 CONF_SAVE = "save"
 
 wifi_ns = cg.esphome_ns.namespace("wifi")
@@ -179,8 +179,8 @@ WIFI_NETWORK_STA = WIFI_NETWORK_BASE.extend(
 def validate_variant(_):
     if CORE.is_esp32:
         variant = get_esp32_variant()
-        if variant in NO_WIFI_VARIANTS:
-            raise cv.Invalid(f"{variant} does not support WiFi")
+        if variant in NO_WIFI_VARIANTS and "esp32_hosted" not in fv.full_config.get():
+            raise cv.Invalid(f"WiFi requires component esp32_hosted on {variant}")
 
 
 def final_validate(config):

